API/Controllers/UserController.cs

```diff
@@ -166,5 +172,44 @@ public async Task<IActionResult> DeleteAccount(int userId)
             userService.Save();
             return Ok();
         }
+
+        /// <summary>
+        ///     Delete all user data
+        /// </summary>
+        /// <returns></returns>
+        [HttpDelete("{userId}/allData")]
+        [Authorize(Policy = nameof(Defaults.Scopes.UserWrite))]
+        public async Task<IActionResult> DeleteAllUserData(int userId)
+        {
+            if(await userService.FindAsync(userId) == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed getting the user account.",
+                    Detail = "The database does not contain a user with this student id.",
+                    Instance = "TODO-CHANGE-TO-GENERATED-INSTANCE-CODE"
+                };
+                return NotFound(problem);
+            }
+            List<Project> projects = (List<Project>) await projectService.GetAllWithUsersAsync();
+            projects.ForEach(delegate(Project project) {
+                if(project.UserId.Equals(userId))
+                {
+                    project.UserId = -1;
+                    project.User = null;
+                    projectService.Update(project);
```

Comment:
In my opinion this (whole updating project section) should be moved into the Logic layer. In the future we might want to reassign the project to some other collaborator (which would be a new business rule).

Reply:
Jurjen and I discussed whether to put it in the controllers or logic, and for now the controller seemed the most obvious place as we can use the existing servicemethods. On the other hand it also makes more sense to put it in the logic layer. Can we discuss this next monday?

Reply:
We can discuss it in here, please contact me if you want to discuss something in voice.